pilot/pkg/xds/workloadentry.go

```diff
@@ -121,6 +130,38 @@ func (sg *InternalGen) QueueUnregisterWorkload(proxy *model.Proxy) {
 	}, features.WorkloadEntryCleanupGracePeriod)
 }
 
+// UpdateWorkloadEntryHealth updates the associated WorkloadEntries health status
+// based on the corresponding health check performed by istio-agent.
+func (sg *InternalGen) UpdateWorkloadEntryHealth(proxy *model.Proxy, event HealthEvent) {
+	if !features.WorkloadEntryHealthChecks {
+		return
+	}
+	// we assume that the workload entry exists
+	entryName := autoregisteredWorkloadEntryName(proxy)
+	if entryName == "" {
+		adsLog.Errorf("unable to derive WorkloadEntry Name, blank name")
+		return
+	}
+
+	// get previous status
+	cfg := sg.Store.Get(gvk.WorkloadEntry, entryName, proxy.Metadata.Namespace)
+	if cfg == nil {
+		adsLog.Debugf("cfg nil when getting WorkloadEntry %v", entryName)
+		return
+	}
+
+	// replace the updated status
+	status := cfg.Status.(*v1alpha1.IstioStatus)
```

Comment:
I think I saw issues when modifying something returned directly from Store. You may need to DeepCopy (which probably doesn't include Status at the moment)